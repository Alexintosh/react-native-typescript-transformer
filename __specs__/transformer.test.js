const file = `
type Cheese = {
  readonly cheese: string
}

export default function Cheese(): Cheese {
  return {cheese: 'stilton'};
}
`

const transformer = require('../')

describe('the transformer', () => {
  it('works', () => {
    const result = transformer.transform(file, 'blah.tsx', {})
    expect(result.code).toMatchSnapshot()
    expect(result.map).toMatchSnapshot()
  })
})
