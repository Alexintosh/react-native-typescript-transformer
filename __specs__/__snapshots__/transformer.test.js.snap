// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the transformer works 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", { value: true });
function Cheese() {
    return { cheese: 'stilton' };
}
exports.default = Cheese;"
`;

exports[`the transformer works 2`] = `
Object {
  "mappings": "AAAA;;AACAA,OAAOC,cAAP,CAAsBC,OAAtB,EAA+B,YAA/B,EAA6C,EAAEC,OAAO,IAAT,EAA7C;AACA,SAASC,MAAT,GAAkB;AACd,WAAO,EAAEC,QAAQ,SAAV,EAAP;AACH;AACDH,QAAQI,OAAR,GAAkBF,MAAlB",
  "names": Array [
    "Object",
    "defineProperty",
    "exports",
    "value",
    "Cheese",
    "cheese",
    "default",
  ],
  "sources": Array [
    "blah.tsx",
  ],
  "sourcesContent": Array [
    "
type Cheese = {
  readonly cheese: string
}

export default function Cheese(): Cheese {
  return {cheese: 'stilton'};
}
",
  ],
  "version": 3,
}
`;
